model/inline/src/main/scala/aqua/model/inline/TagInliner.scala

```diff
@@ -369,23 +338,13 @@ object TagInliner extends Logging {
           }
         } yield model.fold(TagInlined.Empty())(m => TagInlined.Single(model = m))
 
-      case RestrictionTag(name, typ) =>
-        // Rename restriction after children are inlined with new exports
-        TagInlined
-          .After(
-            for {
-              exps <- Exports[S].exports
-              model = exps.get(name).collect { case VarModel(n, _, _) =>
-                RestrictionModel(n, typ)
-              }
-            } yield model.getOrElse(RestrictionModel(name, typ))
-          )
-          .pure
-
       case DeclareStreamTag(value) =>
         value match
-          case VarRaw(name, t) =>
-            Exports[S].resolved(name, VarModel(name, t)).as(TagInlined.Empty())
+          case VarRaw(name, t: StreamType) =>
+            for {
+              _ <- Exports[S].resolved(name, VarModel(name, t))
+              _ <- Exports[S].addStream(name, t)
```

Comment:
`addStream` always comes after `resolved`. Do you have any cases in mind when there will be `addStream` without `resolved`? Maybe lets just embed updating information about streams inside `resolved`?